server/config/raw/global_cfg.go

```diff
@@ -26,6 +26,26 @@ type GlobalCfg struct {
 	Persistence          Persistence          `yaml:"persistence" json:"persistence"`
 	RevisionSetter       RevisionSetter       `yaml:"revision_setter" json:"revision_setter"`
 	Admin                Admin                `yaml:"admin" json:"admin"`
+	TerraformAdminMode   TerraformAdminMode   `yaml:"terraform_admin_mode" json:"terraform_admin_mode"`
+}
+
+type TerraformAdminMode struct {
+	Repo string `yaml:"repo" json:"repo"`
+	Root string `yaml:"root" json:"root"`
+}
+
+func (t TerraformAdminMode) ToValid() valid.TerraformAdminMode {
+	return valid.TerraformAdminMode{
+		Repo: t.Repo,
+		Root: t.Root,
+	}
+}
+
+func (t TerraformAdminMode) Validate() error {
+	return validation.ValidateStruct(&t,
+		validation.Field(&t.Repo),
+		validation.Field(&t.Root),
```

Comment:
did you intend on adding a rule here?

Reply:
I did not. I mean I guess we could but the problem is the string could just be garbage. what are your thoughts?